src/sage/rings/polynomial/polynomial_ring.py

```diff
@@ -1776,7 +1777,38 @@ def monics( self, of_degree=None, max_degree=None ):
         raise ValueError("you should pass exactly one of of_degree and max_degree")
 
 
-class PolynomialRing_commutative(PolynomialRing_general):
+# Placeholder class for deprecation
+class PolynomialRing_general(PolynomialRing_generic):
```

Comment:
This idea of creating a deprecated subclass for the old name does not work because it breaks user code that uses `isinstance(...,  PolynomialRing_general)`.

Reply:
Please suggest a solution as well, as I don't know one. I can remove it, since kwankyu is positive on that as well :)

Reply:
If it's just a comment then no users will see it, and if it's deprecated (meaning it will be removed in the future) then might as well remove it now.

Also I don't understand your `isinstance` code that "breaks". This works for me:
```sage
sage: from sage.rings.polynomial.polynomial_ring import PolynomialRing_generic, PolynomialRing_general
sage: R = PolynomialRing_general(QQ, name="a")
<ipython-input-3-1ad511612f9e>:1: DeprecationWarning: The class PolynomialRing_general has been renamed to PolynomialRing_generic.
See https://github.com/sagemath/sage/issues/38207 for details.
  R = PolynomialRing_general(QQ, name="a")
sage: isinstance(R, PolynomialRing_general)
True
sage: isinstance(R, PolynomialRing_generic)
True
```

Reply:
> If it's just a comment then no users will see it,

Except those that read the release notes

Reply:
What breaks is this:
```
sage: R = QQ['a']
sage: isinstance(R, PolynomialRing_general)
```

Reply:
> > If it's just a comment then no users will see it,
> 
> Except those that read the release notes

I would not expect many people to read the release notes, especially for such a technical detail.

I think is is okay for subclassing to break provided the deprecation warning is emitted. We might be forced to accept a less-than-perfect solution here, but as long as it indicates what a user should do to change their code, then it is acceptable to me.

Reply:
> I think is is okay for subclassing to break provided the deprecation warning is emitted.

No. By definition, during the deprecation phase nothing is broken, users just get the warning. 
Deprecation warnings warn about future breakage, not about current breakage.